src/Cli/dotnet/commands/dotnet-test/SolutionAndProjectUtility.cs

```diff
@@ -195,40 +195,64 @@ private static bool IsValidTargetFramework(Project project, string targetFramewo
             }
 
             string targetFramework = project.GetPropertyValue(ProjectProperties.TargetFramework);
-
-            string executablePath = GetExecutablePath(project);
-            string targetPath = !string.IsNullOrEmpty(executablePath) ? executablePath : project.GetPropertyValue(ProjectProperties.TargetPath);
+            TestModuleRunInformation runInfo = GetTestRunInfo(project);
             string projectFullPath = project.GetPropertyValue(ProjectProperties.ProjectFullPath);
             string runSettingsFilePath = project.GetPropertyValue(ProjectProperties.RunSettingsFilePath);
 
-            return new TestModule(targetPath, PathUtility.FixFilePath(projectFullPath), targetFramework, runSettingsFilePath, isTestingPlatformApplication, isTestProject);
-        }
-
-        private static string GetExecutablePath(Project project)
-        {
-            _ = bool.TryParse(project.GetPropertyValue(ProjectProperties.IsExecutable), out bool isExecutable);
-            _ = bool.TryParse(project.GetPropertyValue(ProjectProperties.UseAppHost), out bool useAppHost);
+            return new TestModule(runInfo, PathUtility.FixFilePath(projectFullPath), targetFramework, runSettingsFilePath, isTestingPlatformApplication, isTestProject);
 
-            string targetFrameworkIdentifier = project.GetPropertyValue(ProjectProperties.TargetFrameworkIdentifier);
-
-            if (targetFrameworkIdentifier.Equals(CliConstants.NetCoreIdentifier, StringComparison.OrdinalIgnoreCase) &&
-               isExecutable &&
-               useAppHost)
+            static TestModuleRunInformation GetTestRunInfo(Project project)
```

Comment:
Again, we're doing all of this in `run` already - I'd love to see some common utils/sharing of code here.

Reply:
This is done now.